imblearn/utils/_tags.py

```diff
@@ -0,0 +1,13 @@
+from dataclasses import dataclass
+
+import sklearn
+from sklearn.utils.fixes import parse_version
+
+sklearn_version = parse_version(parse_version(sklearn.__version__).base_version)
+
+if sklearn_version >= parse_version("1.6"):
+    from sklearn.utils._tags import InputTags
```

Comment:
A try / except around the import (from the public path) makes more sense I think, which shows the exact thing you care about, instead of using the version as a proxy.

Reply:
The Tags are not available in a public path if I'm not wrong.

Reply:
Sorry I check the wrong `__init__.py` :)